### PR TITLE
PCIe Virtual Channel support

### PR DESCRIPTION
**Pull request by tbursztyka**
Here is a basic support for enabling VC on an endpoint.

Only hw round robin is accounted for in port arbitration. It is the simplest (and default) one. Others are non-testable atm also.

### CODE REVIEW

File: drivers/pcie/host/vc.h [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+#ifndef ZEPHYR_DRIVERS_PCIE_HOST_VC_H_
+#define ZEPHYR_DRIVERS_PCIE_HOST_VC_H_
+
+#define PCIE_VC_CAP_REG_1_OFFSET	0x04U
+#define PCIE_VC_CAP_REG_2_OFFSET	0x08U
+#define PCIE_VC_CTRL_STATUS_REG_OFFSET	0x0CU
+
+/** Virtual Channel capability and control Registers */
+struct vc_regs {
```
Comment (jhedberg) [inline drivers/pcie/host/vc.h:15]: Should this perhaps follow the same name spacing, i.e. `struct pcie_vc_regs`?

File: drivers/pcie/host/vc.h [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+#define PCIE_VC_PA_RR		BIT(0)
+#define PCIE_VC_PA_WRR		BIT(1)
+#define PCIE_VC_PA_WRR64	BIT(2)
+#define PCIE_VC_PA_WRR128	BIT(3)
+#define PCIE_VC_PA_TMWRR128	BIT(4)
+#define PCIE_VC_PA_WRR256	BIT(5)
+
+/** Virtual Channel Resource Registers */
+struct vc_resource_regs {
```
Comment (jhedberg) [inline drivers/pcie/host/vc.h:71]: Same here

File: drivers/pcie/host/vc.h [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+			uint32_t _reserved2            : 14;
+		};
+		uint32_t raw;
+	} status_reg;
+};
+
+uint32_t vc_cap_lookup(pcie_bdf_t bdf, struct vc_regs *regs);
+
+void vc_load_resources_regs(pcie_bdf_t bdf, uint32_t base,
+			    int nb_regs, struct vc_resource_regs *regs);
```
Comment (jhedberg) [inline drivers/pcie/host/vc.h:124]: And these functions. I realise that these are internal to `drivers/pcie/host/` however (not public header files)

Comment (tbursztyka) [reply drivers/pcie/host/vc.h:124]: could still be prefixed, I have to preference.

**Review by jhedberg — APPROVED** [review]

File: drivers/pcie/host/vc.h [hunk 4]
```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2021 Intel Corporation.
```
Comment (finikorg) [inline drivers/pcie/host/vc.h:2]: 2023 ?

**Review by finikorg — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

File: drivers/pcie/host/vc.h (hunk 4 above)
Comment (tbursztyka) [reply drivers/pcie/host/vc.h:2]: good catch, I missed it, will send a PR for it

**Pull request merged**